test/suite/test_timestamp09.py

```diff
@@ -62,7 +62,6 @@ def test_timestamp_api(self):
             lambda: self.session.timestamp_transaction(
                 'commit_timestamp=' + timestamp_str(2)),
                 '/older than the first commit timestamp/')
-        c[2] = 2
```

Comment:
Why are these changes necessary? What was the test doing and why was it wrong?

Reply:
Setting the timestamp is a "hard" failure, that is, it sets WT_TXN_ERROR. Updating a key/value after that failure hits the assert I added in DIAGNOSTIC mode, that we don't try operations after a transaction has failed. So, this change is required by my other changes.

Reply:
Instead of removing, would it be useful to change the failure from an assertion to an error and test getting the error here? I realize that an error is sometimes less useful than an error. I don't know if there is an easy way in the python test to know if we're diagnostic or not and only do the test if not diagnostic.

Reply:
I don't think this tests anything particularly interesting. We could certainly add tests that we catch attempts to perform additional operations after a transactional failure, for a variety of methods. I thought about that, but didn't because I didn't think it would add a lot of value and it would be a chunk of work.

Reply:
Then that is fine with me. If it was easy, that would be good. If it is a chunk of work then removal of the line is good with me.